cmd/sso.go

```diff
@@ -0,0 +1,259 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+
+	"github.com/supabase/cli/internal/sso/create"
+	"github.com/supabase/cli/internal/sso/get"
+	"github.com/supabase/cli/internal/sso/list"
+	"github.com/supabase/cli/internal/sso/remove"
+	"github.com/supabase/cli/internal/sso/update"
+	"github.com/supabase/cli/internal/utils"
+	"github.com/supabase/cli/pkg/api"
+)
+
+var (
+	ssoCmd = &cobra.Command{
+		GroupID: groupManagementAPI,
+		Use:     "sso",
+		Short:   "Manage Single Sign-On (SSO) authentication for projects",
+	}
+
+	ssoProjectRef           string
+	ssoMetadataFile         string
+	ssoMetadataURL          string
+	ssoMetadata             bool
+	ssoAttributeMappingFile string
+	ssoDomains              []string
+	ssoAddDomains           []string
+	ssoRemoveDomains        []string
+	ssoOutput               = utils.OutputFlag(false /* allow env output */)
+
+	ssoAddCmd = &cobra.Command{
+		Use:     "add <type = saml> [flags]",
+		Short:   "Add a new SSO identity provider",
+		Args:    cobra.ExactArgs(1),
+		Example: `  supabase sso add saml --project-ref abcdefghijklmn --metadata-file ~/SAMLMetadata.xml`,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(cmd.MarkFlagRequired("project-ref"))
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var params api.CreateProviderForProjectJSONRequestBody
+
+			switch strings.ToLower(args[0]) {
+			case "saml":
+				params.Type = api.Saml
+
+			default:
+				return errors.New("type must be saml")
+			}
+
+			fsys := afero.NewOsFs()
+
+			if ssoMetadataFile != "" {
+				data, err := readMetadataFile(fsys, ssoMetadataFile)
+				if err != nil {
+					return err
+				}
+
+				params.MetadataXml = &data
+			} else if ssoMetadataURL != "" {
+				// TODO: fetch and validate Metadata
```

Comment:
hmm do we need to validate this or can we just leave it to gotrue?

Reply:
Nothing too complex, but fetching the URL, checking it's returning 200 and XML should be enough. Also that it's HTTPS.